internal/pprocess/cleaner/cleaner.go

```diff
@@ -0,0 +1,89 @@
+package cleaner
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"gopkg.in/reform.v1"
+
+	"github.com/percona/pmm-managed/internal/pprocess"
+)
+
+// Cleaner holds unexported field of the background cleaner process.
+type Cleaner struct {
+	db       *reform.DB
+	ticker   chan time.Time
+	stopChan chan struct{}
+	lock     *sync.Mutex
+	running  bool
+}
+
+// New returns a new instance of the cleaner object that implements the PProcess interface.
+func New(db *reform.DB, ticker chan time.Time) pprocess.PProcess {
+	return &Cleaner{
+		db:       db,
+		ticker:   ticker,
+		stopChan: make(chan struct{}),
+		lock:     &sync.Mutex{},
+	}
+}
+
+// Start starts the background process cleaner.
+func (p *Cleaner) Start(ctx context.Context) error {
+	if p.isRunning() {
+		return fmt.Errorf("cleaner process already started")
+	}
+	go p.run(ctx)
+	return nil
+}
+
+// Stop the background process.
+func (p *Cleaner) Stop() error {
+	if !p.isRunning() {
+		return fmt.Errorf("cleaner process is not running")
+	}
+
+	p.stopChan <- struct{}{}
+	return nil
+}
+
+func (p *Cleaner) run(ctx context.Context) {
```

Comment:
I think we can remove `Start`, `Stop` methods and use only this one.

Reply:
If we need it to be run in background we are able to call it in separate goroutine.
And if we need to stop it we can stop it by cancelling context.